### DISCUSSION TITLE
Is it possible to set a default max-age or expires header for staticfiles?

### DISCUSSION THREAD

**Discussion by johnraz** [encode/starlette#1779] [Potential Issue] [answered]
Hi 👋 

I'm currently using `StaticFiles` behind AWS CloudFront.

CloudFront expects either the `Expires` or `Cache-Control: max-age` headers to be set in order to decide for how long something should go in the cache before trying to invalidate it.

If those headers are not set it will default to some configured default value, so I'm not really blocked here.

Looking at the code I couldn't find any clear way to add either the `Expires` or `Cache-Control: max-age` headers when using the `StaticFiles` feature.

Am I missing something?
I think it would be a nice addition for use-case where there is no actual caching-proxy between the starlette app and the client.

Thanks a lot!

**Comment by Kludex** [comment]
@abersheeran I have the feeling you can help here. Am I right? 👀

(I'm away from the computer for some days)

**Comment by abersheeran** [comment, marked as answer]
Starlette does not currently support configuring them. But you can use https://baize.aber.sh/asgi#files, as a standard ASGI Application, you can use it directly in starlette project (Just replace `StaticFiles`).

**Reply by Kludex** [reply]
Thanks @abersheeran 🙏

**Comment by johnraz** [comment]
awesome @Kludex @abersheeran  thanks a lot 👍
